packages/rich-text/src/to-html-string.js

```diff
@@ -40,7 +40,16 @@ export function toHTMLString( { value, preserveWhiteSpace } ) {
 		appendText,
 	} );
 
-	return createChildrenHTML( tree.children );
+	const string = new String( createChildrenHTML( tree.children ) );
+
+	Object.defineProperty( string, 'value', {
+		value,
+	} );
+	Object.defineProperty( string, 'preserveWhiteSpace', {
+		value: preserveWhiteSpace,
+	} );
+
+	return string;
```

Comment:
I guess our intent here is to make sure that this still passes `instanceof String` checks, but we're now setting properties on every object instance itself, vs. having a single property defined on the class.

I'd be curious to know what impact this has on the runtime with regards to performance. probably nothing, but would a subclass of `String` still serve?

We'd still have to store these property values, so maybe it's a wash, but it feels odd to see us defining properties on instances of the `String` class.

Reply:
We could try a sub class later. The point is that the value in the store becomes an object in which we can store any information.

Reply:
sure, but why not do this instead?

```js
String.prototype.richTextValue = function() {
	return this.richTextValue;
}
```

or just this…
```
const string = new String( … );
string.value = value;
```

the sub-class idea is just a way to differentiate without per-instance property defines.

```js
class RichTextString extends String {
	public function construct( initialValue ) {
		this.value = initialValue;

		…
	}

	public function value() {
		return this.value;
	}
}
```

not a biggie, just noted the oddity.

Reply:
> or just this…
> 
> ```
> const string = new String( … );
> string.value = value;
> ```

This is what we're doing, no?

Reply:
> This is what we're doing, no?

literally no, from the code. I guess the main difference here is that we're creating a read-only non-enumerable property on `string` instead of assigning a value.

`value` here doesn't show up in `Object.keys(string)`

it was the unexplained `Object.define()` I was asking mostly about, because it looks clear that we specifically don't want to use assignment, but not clear what our objective is. _why_ not assign `string.value = value`

Reply:
Resolving this since we moved on to a class.